warehouse/query-core/src/test/java/datawave/query/util/sortedset/BufferedFileBackedSortedSetTest.java

```diff
@@ -0,0 +1,233 @@
+        int[] sortedTemplate = new int[] {1, 5, 3, 0, 8, 10, 4, 11, 9, 7, 6, 2};
+        data = new byte[template.length * 2][];
+        for (int i = 0; i < template.length; i++) {
+            data[i] = new byte[i + 11];
+            Arrays.fill(data[i], template[i]);
+        }
+        for (int i = 0; i < template.length; i++) {
+            data[i + template.length] = new byte[10];
+            Arrays.fill(data[i + template.length], template[i]);
+        }
+        sortedOrder = new int[data.length];
+        for (int i = 0; i < template.length; i++) {
+            sortedOrder[i * 2] = sortedTemplate[i] + sortedTemplate.length;
+            sortedOrder[i * 2 + 1] = sortedTemplate[i];
+        }
+        set = new BufferedFileBackedSortedSet<>(new ByteArrayComparator(), 5, 7, new BufferedFileBackedSortedSet.SortedSetFileHandlerFactory() {
+            @Override
+            public FileSortedSet.SortedSetFileHandler createHandler() throws IOException {
+                return new SortedSetTempFileHandler();
+            }
+        });
+        
+        // adding in the data set multiple times to create underlying files with dyuplicate values making the
+        // MergeSortIterator's job a little tougher...
+        for (int d = 0; d < 11; d++) {
+            Collections.addAll(set, data);
+        }
+    }
+    
+    @After
+    public void tearDown() throws Exception {
+        data = null;
+        sortedOrder = null;
+        set.clear();
+        set = null;
+    }
+    
+    @Test
+    public void testSize() {
+        int expectedSize = data.length;
+        assertEquals(expectedSize, set.size());
+        for (int i = (data.length / 2); i < data.length; i++) {
+            set.remove(data[i]);
+            expectedSize--;
+            assertEquals(expectedSize, set.size());
+        }
+        for (int i = 0; i < (data.length / 2); i++) {
+            set.remove(data[i]);
+            expectedSize--;
+            assertEquals(expectedSize, set.size());
+        }
+        assertEquals(0, set.size());
+        for (int i = 0; i < data.length; i++) {
+            set.add(data[i]);
+            expectedSize++;
+            assertEquals(expectedSize, set.size());
+        }
+    }
+    
+    @Test
+    public void testIsEmpty() {
+        assertFalse(set.isEmpty());
+        for (int i = (data.length / 2); i < data.length; i++) {
+            set.remove(data[i]);
+            assertFalse(set.isEmpty());
+        }
+        for (int i = 1; i < (data.length / 2); i++) {
+            set.remove(data[i]);
+            assertFalse(set.isEmpty());
+        }
+        set.remove(data[0]);
+        assertTrue(set.isEmpty());
+        for (int i = 0; i < data.length; i++) {
+            set.add(data[i]);
+            assertFalse(set.isEmpty());
+        }
+    }
+    
+    @Test
+    public void testClear() {
+        set.clear();
+        assertTrue(set.isEmpty());
+        for (int i = 0; i < data.length; i++) {
```

Comment:
This part of the test duplicates the test above. I think we can drop this for loop portion.